src/Components/Web.JS/src/InputFile.ts

```diff
@@ -117,21 +121,22 @@ export function getFileById(elem: InputElement, fileId: number): BrowserFile {
 }
 
 function getArrayBufferFromFileAsync(elem: InputElement, fileId: number): Promise<ArrayBuffer> {
-  const file = getFileById(elem, fileId);
-
-  // On the first read, convert the FileReader into a Promise<ArrayBuffer>.
-  if (!file.readPromise) {
-    file.readPromise = new Promise(function(resolve: (buffer: ArrayBuffer) => void, reject): void {
-      const reader = new FileReader();
-      reader.onload = function(): void {
-        resolve(reader.result as ArrayBuffer);
-      };
-      reader.onerror = function(err): void {
-        reject(err);
-      };
-      reader.readAsArrayBuffer(file['blob']);
```

Comment:
Can't allocate a buffer larger than ~2GB so this'll fail for large files.

Reply:
Added this back in for now so this PR isn't blocked on WASM streaming interop. Will remove it in that PR.

Reply:
> Added this back in for now so this PR isn't blocked on WASM streaming interop. Will remove it in that PR.

Are you saying that, as of this PR, we still won't support >2GB files but it will *nearly* support it and then when you do the WASM piece it will support it on Server too?

If so that's totally fine - just want to check I understand.

Reply:
As of this PR we'll support >2GB in server and up to the current ~2GB limit in WASM. With the new WASM PR, I have >2GB working in WASM as well (using this `Blob.slice` approach).

Reply:
@TanayParikh where can I find this "the new WASM PR" and information weather Blazor WASAM 6.0 supports >2GB upload or not?

I can't find any documentation or examples on uploading >2GB file in WASM to an API (preferably with progressive information for UI display)? The docs on [Blazor file upload](https://docs.microsoft.com/en-us/aspnet/core/blazor/file-uploads?view=aspnetcore-5.0&pivots=webassembly) doesn´t event mention the <2GB limit

Its **totally** unclear if this works now in .net 6.0 , just like I think you are pointing out [here](https://github.com/dotnet/aspnetcore/issues/33638#issuecomment-880389920). 

The internet is littered with questions like [this](https://stackoverflow.com/questions/66770670/streaming-large-files-from-blazor-webassembly) and [this](https://stackoverflow.com/a/62023601/1187583) and no answers. 

Hope you can point me in the right direction.

Reply:
> where can I find this "the new WASM PR"

https://github.com/dotnet/aspnetcore/pull/33986

> weather Blazor WASAM 6.0 supports >2GB upload or not?

It does.

> I can't find any documentation or examples on uploading >2GB file in WASM to an API (preferably with progressive information for UI display)?

The process is the same as you would take to upload files < 2GB. We recently added docs for showing progress for uploads in Blazor server [(here)](https://docs.microsoft.com/en-us/aspnet/core/blazor/file-uploads?view=aspnetcore-6.0&pivots=server#upload-files-with-progress), you should be able to use that as guidance for implementing something similar in Blazor WASM.

> The docs on [Blazor file upload](https://docs.microsoft.com/en-us/aspnet/core/blazor/file-uploads?view=aspnetcore-5.0&pivots=webassembly) doesn´t event mention the <2GB limit

The 2GB limit in .NET 5 and before was un-intended and un-documented behavior. This has been resolved in .NET 6. I've created an issue to add this info to this limitation in .NET 5 to the docs. https://github.com/dotnet/AspNetCore.Docs/issues/24381

Reply:
Awesome! Thanks for the update. This helped me allot deciding on my next steps